samples/boards/rpi_pico/uart_pio/boards/rpi_pico.overlay

```diff
@@ -0,0 +1,30 @@
+&pinctrl {
+	pio1_uart_default: pio1_uart_default {
+		pio1_uart_rx: group1 {
+			#pin-cells = <0>;
+			pinmux = <PIO1_P1>;
+			input-enable;
+			bias-pull-up;
+		};
+		pio1_uart_tx: group2 {
+			#pin-cells = <0>;
+			pinmux = <PIO1_P0>;
+		};
+	};
+};
+
+&pio1 {
+ 	status = "okay";
+
+	pinctrl-0 = <&pio1_uart_default>;
+	pinctrl-names = "default";
+	state-machine-num = <2>;
+
+	pins = <&pio1_uart_rx>, <&pio1_uart_tx>;
```

Comment:
Need to specify which index of the pinmux array.
I tried it and it seems to work with the attached code.

[0001-add-pinmux-cell.patch](https://github.com/zephyrproject-rtos/zephyr/files/10902389/0001-add-pinmux-cell.patch)

Reply:
Why is the index needed?

Reply:
The `pinmux` property is an array. There is a possibility to use the second or later elements.

e. g. SPI's pinctrl is usually grouped into OUTPUT and INPUT.
If we implement SPI with PIO, we use all of the OUTPUT group's elements.

https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/arm/rpi_pico/rpi_pico-pinctrl.dtsi#L27-L35